app.rb

```diff
@@ -38,49 +44,18 @@ def create_person
     type = gets.chomp.to_i
 
     if type == 1
-      student = create_student
+      student = @person_factory.create_student
       @people << student unless student.nil?
       puts "Student #{student.name} created."
     elsif type == 2
-      teacher = create_teacher
+      teacher = @person_factory.create_teacher
       @people << teacher unless teacher.nil?
       puts "Teacher #{teacher.name} created."
     else
       puts 'Invalid choice,person not created.'
     end
```

Comment:
- [ ] Dear @Wils-FOURR-speed, I notice you have created a separate method for `list_books` and `list_people`, which is great ✅ . Now following up on that, for creating a person, the first function better be only responsible for checking the `type` of person we are creating, we can then have two methods, `create_student` and `create_teacher` both will be solely responsible for creating a `person` or a `teacher` accordingly ⚠️ . This is following up a Single responsibility of the SOLID principle.